src/cpp/session/SessionSuspend.cpp

```diff
@@ -486,9 +487,6 @@ void checkForSuspend(const boost::function<bool()>& allowSuspend)
       // errors will be logged/reported internally and we will move on
       suspendSession(false);
    }
-
-   // timeout suspend
-   checkForTimeout(allowSuspend);
```

Comment:
It's been so long, I can't remember exactly, but moving this from the beginning of the `waitForMethod` loop to the end is raising a red flag in my head. I vaguely recall hitting issues when I moved this line around.

If you've been testing this locally and it seems to be running fine then maybe we're okay, but once this builds I'll want to run through some tests with it.

Reply:
Yes, it seems to suspend an idle session just fine and I've tested one of the ways to block suspension. I could try more ways to block suspension, but we didn't really change that logic.  I also used temporary debug messages to be sure that key canSuspend() is called at the right times and with a lower volume.